### DISCUSSION TITLE
submit Pekko fork to Apache Software Foundation

### DISCUSSION THREAD

**Discussion by pjfanning** [matsluni/aws-spi-akka-http#238] [General]
Hi @matsluni 

We use a fork of this project (https://github.com/pjfanning/aws-spi-pekko-http) as a dependency in [Apache Pekko Connectors](https://github.com/apache/incubator-pekko-connectors). We are grateful for the use of your code in a number of our connectors. 

There is discussion about adding this to the code to the apache/incubator-pekko-connectors repo itself so that we lower the number of non-Apache dependencies.

To this, ideally, we would get you to grant the code to the ASF. This doesn't stop you from continuing with this repo but it gives us the legal basis to bring a fork of your code in the ASF. 

The form and the process for submitting it are detailed at https://www.apache.org/licenses/contributor-agreements.html#grants

**Comment by matsluni** [comment]
Hi @pjfanning,

that sounds like a good idea to me. 

How about the code contributed by others via PRs? 

Let me have a look at the link you mentioned below and get back to you.

**Reply by pjfanning** [reply]
If there are particular users who have contributed a lot of code, we can follow up with them too. In the real world, we wouldn't ask everyone's permission, just the biggest contributors. We aren't even changing the existing license - still Apache licensing. The main diff is that if we get a software grant from you, we would change the code headers to be standard Apache headers - so Matthias Lüneberg copyright would no longer appear. We would acknowledge the origin of the code in our NOTICE file though.

**Reply by mdedetrich** [reply]
I just manually went through the git log and by far the majority of the commits came through @matsluni or @scala-steward . Everyone else just contributed 1-2 commits at most and most of the commits were quite trivial

@matsluni I think its fine to proceed, all that we need from you is to sign an [SGA](https://www.apache.org/licenses/contributor-agreements.html#grants) and we can continue through the process.

**Reply by matsluni** [reply]
Hi @pjfanning and @mdedetrich, I just sent the signed copy of the software grant to the apache foundation. Please let me know in case anything is missing.

**Reply by mdedetrich** [reply]
@matsluni So I can confirm that we have received the SGA which means we can proceed. There are 2 ways to do this, one is that you create the PR against https://github.com/apache/incubator-pekko-connectors or I make the PR but give you authorship rights via the `Co-Authored-By` tag in the git commit message. If I end up doing it I will loop you in as a reviewer just to make sure that everything is okay but since the code is quite trivial I am not expecting any problems

What I had in mind is that we would create a new sbt project/maven artifact within sbt and then have

- `AwsLambda`
- `DynamoDB`
- `Kinesis`
- `Eventbridge`
- `Sns`
- `Sqs`

All depend on it via `dependsOn`. `aws-spi-pekko-http` will also be published as an artifact, it just will be published alongside with everything else in pekko-connectors.

**Reply by pjfanning** [reply]
@mdedetrich You probably want the code from https://github.com/pjfanning/aws-spi-pekko-http as opposed to taking the code directly from this repo.
We are also not through all the ASF hoops yet. We will need to get IP clearance first. See https://incubator.apache.org/guides/ip_clearance.html and https://incubator.apache.org/ip-clearance/pekko-http-cors.html (last Pekko one).

**Reply by pjfanning** [reply]
I have created https://incubator.apache.org/ip-clearance/pekko-connectors-aws-spi.html and will bring a vote to the Incubator team in a few days.
